tensorflow-ops/src/TensorFlow/Gradient.hs

```diff
@@ -443,6 +444,36 @@ opGrad "Neg" _ [_] [dz] = [Just $ negate $ expr dz]
 opGrad "Relu" _ [toT -> x] [dz] = [Just $ reluGrad dz x]
 opGrad "ReluGrad" _ [_, toT -> x ] [dz] = [Just $ reluGrad dz x, Just $ CoreOps.zerosLike x]
 
+-- Concat concatenates input tensors
+--   x1 of shape s1 = [d1, ..., di_1, ..., dn]
+--   x2 of shape s2 = [d1, ..., di_2, ..., dn]
+--    .           .     .          .        .
+--    .           .     .          .        .
+--    .           .     .          .        .
+--   xm of shape sm = [d1, ..., di_m, ..., dn]
+--  along dimension i to an output tensor
+--   y  of shape sy = [d1, ..., d, ..., dn]
+--  where d = sum di = sum [di_1,...,di_m]
+--
+--  The incoming gradient from backpropagation is
+--   simply forwarded split across input tensors.
+--   Forwarded gradients have shapes s = [s1, ..., sm].
+opGrad "Concat" _ _ix [dy]
```

Comment:
Ok. I've tried to respect this in my latest changes. Is your whitespace convention documented somewhere?